internal/process/provisioning/get_kubeconfig_step.go

```diff
@@ -18,16 +16,13 @@ type GetKubeconfigStep struct {
 	provisionerClient   provisioner.Client
 	operationManager    *process.OperationManager
 	provisioningTimeout time.Duration
-	k8sClientProvider   func(kubeconfig string) (client.Client, error)
 }
 
 func NewGetKubeconfigStep(os storage.Operations,
```

Comment:
Get kubeconfig only gets kubconfig from provisioner and store it in the operation. No need to create (and store) k8s client. The client will be created by the provider when it will be needed

Reply:
This is the only place , where we call Provisioner because not all environments has enabled Infrastructure Manager. In case of Infrastructure Manager is disabled, KEB gets the kubeconfig and stores it in the secret